### PR TITLE
Add enhanced Gold generation in Mesa-Type Biomes

### PR DESCRIPTION
**Pull request by 12xx12**
Gold is generated more often in mesa type biomes - fixed that as discussed in the emerald generation pull request

### CODE REVIEW

File: Server/Plugins/APIDump/APIDesc.lua [hunk 1]
```diff
@@ -13647,6 +13647,23 @@ end
...
+							Type = "EMCSBiome",
+						}
+					},
+					Returns =
+					{
+						{
+							Type = "boolean",
+						},
+					},
+					Notes = "Returns true if the biome is Mesa-Type (mutations of the Mesa biome)."
```
Comment (tigerw) [inline Server/Plugins/APIDump/APIDesc.lua:13665]: "...is a type of Mesa (mutations..." I guess?

File: src/BiomeDef.h [hunk 2]
```diff
@@ -154,6 +154,9 @@ extern bool IsBiomeCold(EMCSBiome a_Biome);
 /** Returns true if the biome is a mountain type */
 extern bool IsBiomeMountain(EMCSBiome a_Biome);
 
+/** Return true if the biome is Mesa or one of its mutation. */
```
Comment (tigerw) [inline src/BiomeDef.h:157]: mutations

File: src/Generating/FinishGen.cpp [hunk 3]
```diff
@@ -1769,6 +1769,25 @@ void cFinishGenOreNests::GenerateOre(
...
+	{
+		auto BiomeSampleOne =    a_ChunkDesc.GetBiome( 4,  4);
+		auto BiomeSampleTwo =    a_ChunkDesc.GetBiome( 4, 12);
+		auto BiomeSampleThree =  a_ChunkDesc.GetBiome(12,  4);
+		auto BiomeSampleFour =   a_ChunkDesc.GetBiome(12, 12);
+
+		if ((IsBiomeMesa(BiomeSampleOne)) ||
+			(IsBiomeMesa(BiomeSampleTwo)) ||
+			(IsBiomeMesa(BiomeSampleThree)) ||
+			(IsBiomeMesa(BiomeSampleFour)))
```
Comment (tigerw) [inline src/Generating/FinishGen.cpp:1784]: I guess you could format it like

```
if (
    A ||
    B ||
)
```

with the trailing ) on another line. (actually same for line 1766)

Also it looks like there's extra parentheses around the individual conditions.

**Comment by tigerw** [comment]
yeah just style

File: src/BiomeDef.h (hunk 2 above)
Comment (tigerw) [reply src/BiomeDef.h:157]: Returns

**Comment by 12xx12** [comment]
Gonna fix this tomorrow

I should write more English -.-

**Comment by tigerw** [comment]
Thanks! And your English is fine :)

File: src/Generating/FinishGen.cpp (hunk 3 above)
Comment (12xx12) [reply src/Generating/FinishGen.cpp:1784]: There is some extra Paranthesis

I thought the styleguide in contributing.md said that

Comment (madmaxoft) [reply src/Generating/FinishGen.cpp:1784]: The parentheses should be around comparisons, not around simple function calls. So it's 
```cpp
if ((A < B) || (B < C))
```
and
```cpp
if (FunctionCall(A) || FunctionCall(B))
```

File: src/BiomeDef.h (hunk 2 above)
Comment (12xx12) [reply src/BiomeDef.h:157]: done

File: Server/Plugins/APIDump/APIDesc.lua (hunk 1 above)
Comment (12xx12) [reply Server/Plugins/APIDump/APIDesc.lua:13665]: also done

File: src/Generating/FinishGen.cpp (hunk 3 above)
Comment (12xx12) [reply src/Generating/FinishGen.cpp:1784]: i also messed up this in the emerald commit so new pull reqeust

**Pull request merged**